### PR TITLE
Update mlflow server model-registry-uri param doc

### PR DESCRIPTION
**Pull request by WeichenXu123**
Signed-off-by: Weichen Xu <weichen.xu@databricks.com>

## Related Issues/PRs

Follow-up fixes for https://github.com/mlflow/mlflow/pull/6142

## What changes are proposed in this pull request?

Update mlflow server model-registry-uri param doc

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/server/__init__.py [hunk 1]
```diff
@@ -17,7 +17,6 @@
 # NB: These are internal environment variables used for communication between
 # the cli and the forked gunicorn processes.
 BACKEND_STORE_URI_ENV_VAR = "_MLFLOW_SERVER_FILE_STORE"
-REGISTRY_STORE_URI_ENV_VAR = "_MLFLOW_SERVER_REGISTRY_STORE"
```
Comment (harupy) [inline mlflow/server/__init__.py:20]: WHat is the purpose of this change?

Comment (WeichenXu123) [reply mlflow/server/__init__.py:20]: looks like this env is designed to be only used for mlflow server. Will revert the change.

File: mlflow/cli.py [hunk 2]
```diff
@@ -255,9 +255,8 @@ def _validate_server_args(gunicorn_opts=None, workers=None, waitress_opts=None):
     default=None,
     help="URI to which to persist registered models. Acceptable URIs are "
     "SQLAlchemy-compatible database connection strings "
-    "(e.g. 'sqlite:///path/to/file.db') or local filesystem URIs "
-    "(e.g. 'file:///absolute/path/to/directory'). By default, data will be logged "
-    "same to 'backend-store-uri'",
+    "(e.g. 'sqlite:///path/to/file.db')."
```
Comment (harupy) [inline mlflow/cli.py:258]:
```suggestion
    "(e.g. 'sqlite:///path/to/file.db'). "
```

File: mlflow/cli.py [hunk 3]
```diff
@@ -255,9 +255,8 @@ def _validate_server_args(gunicorn_opts=None, workers=None, waitress_opts=None):
...
+    "By default, data will be logged same to 'backend-store-uri'",
```
Comment (harupy) [inline mlflow/cli.py:259]:
```suggestion
    "If not specified, `backend-store-uri` is used.",
```

File: mlflow/cli.py [hunk 4]
```diff
@@ -255,9 +255,8 @@ def _validate_server_args(gunicorn_opts=None, workers=None, waitress_opts=None):
     default=None,
     help="URI to which to persist registered models. Acceptable URIs are "
     "SQLAlchemy-compatible database connection strings "
-    "(e.g. 'sqlite:///path/to/file.db') or local filesystem URIs "
-    "(e.g. 'file:///absolute/path/to/directory'). By default, data will be logged "
-    "same to 'backend-store-uri'",
+    "(e.g. 'sqlite:///path/to/file.db'). "
+    "If not specified, `backend-store-uri` is used.",
```
Comment (harupy) [inline mlflow/cli.py:259]:
```
    help="URI to which to persist registered models. Acceptable URIs are "
    "SQLAlchemy-compatible database connection strings (e.g. 'sqlite:///path/to/file.db'). "
    "If not specified, `backend-store-uri` is used.",
```

**Comment by harupy** [comment]
https://github.com/mlflow/mlflow/blob/ca07120386b4866c3f91c2c2e07f9b62f9b9e331/mlflow/cli.py#L367 also needs to be fixed.

**Review by harupy — APPROVED** [review]
